src/Common/Extensions/FileExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Oryx.Common.Extensions
+{
+    /// <summary>
+    /// A set of extension methods to help with operations involving the file system.
+    /// </summary>
+    public static class FileExtensions
+    {
+        /// <summary>
+        /// This method will write the given contents to the path provided and create any non-existent directories
+        /// along the path since File.WriteAllText will throw if one of the directories in the path does not exist.
+        /// </summary>
+        /// <param name="outputPath">The file path to write the contents to.</param>
+        /// <param name="contents">The contents to be written to the given file path.</param>
+        public static void SafeWriteAllText(this string outputPath, string contents)
+        {
+            if (string.IsNullOrEmpty(outputPath))
+            {
+                return;
+            }
+
+            var trimmedOutputPath = outputPath.TrimEnd('/').TrimEnd('\\');
+            var parentPath = Directory.GetParent(trimmedOutputPath).FullName;
+            if (!Directory.Exists(parentPath))
+            {
+                Directory.CreateDirectory(parentPath);
+            }
+
+            File.WriteAllText(outputPath, contents);
+        }
+
+        /// <summary>
+        /// This method will recursively enumerate the files under a given path since the
+        /// Directory.EnumerateFiles call does not check to see if a directory exists before
```

Comment:
I am curious when does this actually happen. I would argue it is the job of this .NET Core api to do these checks and give us the results. If we want to do safe enumeration, it should be enough to just check if supplied argument's directory exists?